class.jetpack.php

```diff
@@ -1855,6 +1855,12 @@ public static function get_active_modules() {
 		} else {
 			$active = array_diff( $active, array( 'vaultpress' ) );
 		}
+		
+		//If protect is active on the main site of a multisite, it should be active on all sites.
+		if ( !in_array( 'protect', $active ) && is_multisite() && get_site_option( 'jetpack_protect_active' ) ) {
+			$active[] = 'protect';
+		}
+		
```

Comment:
This is not a great solution. Because when I'm on a sub-site, I'm still able to click the 'Deactivate' button for the Protect module, and i'm shown a shiny notification telling me it's been deactivated, while it is very clearly still active.

![screen shot 2015-06-15 at 10 33 37 am](https://cloud.githubusercontent.com/assets/2694219/8162267/2c6d6c24-134a-11e5-9c84-d32ffd297a0d.png)

If we are going to force activation on all subsites, we need to address that weird UI.

My suggestion is we allow activation / deactivation on sub-sites, because freedom and america.

BUT we show a warning, like you say, when a user activates on a sub-site but Protect is not activate on the main site. Something like: "Warning: your site will not be fully protected until you activate this module on the main site"